### PR TITLE
Change Namespaces from string[] to string

### PR DESCRIPTION
**Pull request by zdtsw**
## Description
it is defined as `var dscApplicationsNamespace string` which does not really need array for application namespace, only one namespace is enough.
The change is change array to string 
also will support user config namespace they want to use to install apps in.

## How Has This Been Tested?
`make manifests bundle `
cleaned up deepcopy functions

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: apis/dscinitialization/v1alpha1/dscinitialization_types.go [hunk 1]
```diff
@@ -32,8 +32,10 @@ const (
 
 // DSCInitializationSpec defines the desired state of DSCInitialization
 type DSCInitializationSpec struct {
-	Namespaces   []string   `json:"namespaces"`
+	// +kubebuilder:default:=opendatahub
+	Namespace    string     `json:"namespace"`
```
Comment (VaishnaviHire) [inline apis/dscinitialization/v1alpha1/dscinitialization_types.go:36]: Should this be named as applicationsNamespace?

Comment (zdtsw) [reply apis/dscinitialization/v1alpha1/dscinitialization_types.go:36]: sure i can rename it

Comment (zdtsw) [reply apis/dscinitialization/v1alpha1/dscinitialization_types.go]: fixed and updated PR description.

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**